### ISSUE TITLE
alg/ensemble,internal/results: dedupe expensive tests

### ISSUE THREAD

**Issue by mmcloughlin** [mmcloughlin/addchain#102]
Both `alg/ensemble` and `internal/results` run expensive tests to confirm that algorithm results have not changed on targets of interest, but with slightly different goals:

* `alg/ensemble`: Ensure chain lengths for _every_ algorithm results are preserved, to verify no unexpected changes.
* `internal/results`: Ensure that our list of best-known chains is up-to-date, so our documentation is right. This checks the exact programs generated, not just their length.

These tests run every algorithm against every target in the results list, so they're expensive at the moment. The logic could be merged into one test.

Probably low priority because these should get a lot faster (after #60) and they're only run in long or stress test mode right now.
